Cython/Compiler/Annotate.py

```diff
@@ -315,7 +315,7 @@ def annotate(match):
     br'(?P<py_macro_api>Py[A-Z][a-z]+_[A-Z][A-Z_]+)|'
     br'(?P<py_c_api>Py[A-Z][a-z]+_[A-Z][a-z][A-Za-z_]*)'
     br')(?=\()|'       # look-ahead to exclude subsequent '(' from replacement
-    br'(?P<error_goto>(?:(?<=;) *if [^;]* +)?__PYX_ERR\([^)]+\))'
+    br'(?P<error_goto>(?:(?<=;) *if \(((?!if).)*\) +)?__PYX_ERR\([^)]+\))'
```

Comment:
We shouldn't start using positional groups here. And we should make sure that we really find `if …` and not `iffy` etc.
```suggestion
    br'(?P<error_goto>(?:(?<=;) *if \((?:(?!if[ (]).)*\) +)?__PYX_ERR\([^)]+\))'
```
Or could we somehow include HTML escapes explicitly? Are there more than `&amp;` and `&quot;` that can occur here?

Reply:
> We shouldn't start using positional groups here. And we should make sure that we really find if … and not iffy etc.

I agree.

> Or could we somehow include HTML escapes explicitly? Are there more than `&amp;` and `&quot;` that can occur here?

I was thinking about that but I found it rather difficult to express that `&amp;` is allowed but `;` is not. They share the same character `;` which causes it is difficult to express it. Hence, I implemented it by disallowing `if` statement. Maybe it is possible, but I am lacking necessary regex skills to implement that.